core/templates/dev/head/services/improvement-task.service.ts

```diff
@@ -78,6 +78,16 @@ angular.module('oppia').factory('ImprovementTaskService', [
           return [].concat.apply([], tasksFromFactories);
         });
       },
+      getOpenTaskCount: function() {
+        var isTaskOpen = function(task) {
+          return task.getStatus() === 'open';
+        };
+        var fetchedTasks = [];
```

Comment:
This is a good idea -- let's move this to the top of the service so we can cache the value, something like:

```javascript
angular.module('oppia').factory('ImprovementTaskService', [
  /* ... */,
  function(
      /* ... */) {
    // ...

    var tasksPromise = null;
    
    // ...
    
    return {
      // ...

      fetchTasks: function() {
        if (tasksPromise === null) {
          tasksPromise = $q.all(
            improvementTaskObjectFactoryRegistry.map(
              taskFactory => taskFactory.fetchTasks())
            ).then(tasksFromFactories => {
              // Flatten the tasks into a single array before returning.
              return [].concat.apply([], tasksFromFactories);
            });
        }
        return tasksPromise;
      },

      // ...

      getOpenTaskCount: function() {
        return this.fetchTasks().then(allTasks => {
          return allTasks.filter(task => task.isOpen()).length;
        });
      }
  };
}]);